flink-kubernetes-operator/src/test/java/org/apache/flink/kubernetes/operator/controller/FlinkDeploymentControllerTest.java

```diff
@@ -217,9 +227,10 @@ public void verifyBasicReconcileLoop(FlinkVersion flinkVersion) throws Exception
 
     @Test
     public void verifyFailedDeployment() throws Exception {
+
         var submittedEventValidatingResponseProvider =
                 new TestUtils.ValidatingResponseProvider<>(
-                        new EventBuilder().withNewMetadata().endMetadata().build(),
+                        mockedEvent,
```

Comment:
good enough, but can't we send back the the event we received?

Reply:
I tried but the the `TestUtils.ValidatingResponseProvider` class requires the event to be returned at construction time, and I couldn't find a way to intercept the `createOrReplace` call inside `EventUtils.createOrUpdateEvent`